encord/orm/dataset.py

```diff
@@ -323,30 +323,27 @@ class StorageLocation(IntEnum):
     AWS = (1,)
     GCP = (2,)
     AZURE = 3
+    OTC = 4
 
     @staticmethod
     def from_str(string_location: str) -> StorageLocation:
-        if string_location == "CORD_STORAGE":
-            return StorageLocation.CORD_STORAGE
-        if string_location == "AWS_S3":
-            return StorageLocation.AWS
-        if string_location == "GCP_STR":
-            return StorageLocation.GCP
-        if string_location == "AZURE_STR":
-            return StorageLocation.AZURE
-        raise TypeError(f"Invalid storage location string: `{string_location}`")
+        return STORAGE_LOCATION_STR[string_location]
 
     def get_str(self) -> str:
         if self == StorageLocation.CORD_STORAGE:
             return "CORD_STORAGE"
-        if self == StorageLocation.AWS:
+        elif self == StorageLocation.AWS:
             return "AWS_S3"
-        if self == StorageLocation.GCP:
+        elif self == StorageLocation.GCP:
             return "GCP_STR"
-        if self == StorageLocation.AZURE:
+        elif self == StorageLocation.AZURE:
             return "AZURE_STR"
+        elif self == StorageLocation.OTC:
+            return "OTC_STR"
 
 
+STORAGE_LOCATION_STR = {location.get_str(): location for location in StorageLocation}
```

Comment:
Can we do something that this would be typed before runtime?

Reply:
Absolutely. I just came off of writing a bit of typescript which would 100% be able to infer the type here, but I forget mypy is not as capable. Good call :+1:

Reply:
The fact that this: 
![image](https://user-images.githubusercontent.com/86244340/181459755-2c0317c1-d14d-4b1d-8bd0-0ab550b4e3f5.png)
does not get picked up as an error annoys me...